Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/Game.java

```diff
@@ -0,0 +1,115 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.util.ArrayList;
+
+public class Game {
+	protected 	Dictionary 	dictionary;		
+	protected 	Player 		player;		  	
+	protected 	char[] 		etalon;			
+	protected	ArrayList 	<Character> 	guessedLetters;
+	protected	char[] 		openedWord;		
+	protected	ArrayList	<Character>		wrongLetters;
+	protected 	UserInterface ui;			
+	protected 	final int 	TOTAL_GUESSES=9;
+	protected 	int 		currentGuesses;
+
+
+	public Game(){
+		dictionary = new ArrayDictionary ("words.txt");
+		ui = new GUIMainWindow(this);
```

Comment:
Also, if you use your original code of holding a reference to the UI, I'm also not sure why the Game logic gets to decide it should use a GUIMainWindow. Surely Game would rather manipualte some generic UI via an interface, and the caller can decide a suitable UI for its purpose? (i.e. dependency injection)